### ISSUE TITLE
ssl options somewhat confusing

### ISSUE THREAD

**Issue by koertkuipers** [Stratio/spark-mongodb#17]
i find the way ssl is handled somewhat confusing. it seems turning ssl on/off and setting the trust/keystore are conflated.

i would like to have a switch to turn on/off ssl for the mongo client without having to provide the keystore/truststore details. we set these system properties somewhere else in a more central manner.

if it wasnt for backwards compatibility i would suggest to remove keystore and truststore system properties entirely from the scope of this library. that seems a cleaner design to me.

best, koert

**Comment by pmadrigal** [comment]
Hi,

We are thinking to refactor this part, and maybe it is better to offer both options:

- User can set system properties by itself.

- User can pass the ssl parameters and we set the system properties. This option takes precedence if system properties had been configured previously and then, user pass the parameters.

Furthermore, we are valuing add a configuration file to add any configuration parameters from there.

**Comment by koertkuipers** [comment]
That sounds great. I assume in that refactoring you would add a deepconfig
setting to just turn mongo ssl on/off?

About adding a config file: typesafe config is very convenient and
integrates nicely with system properties as well

**Comment by pmadrigal** [comment]
Yes, we will add a on/off ssl parameter, and we will take a look to typesafe config.

It will be available within the next weeks.

Thank you for your feedback! :)

**Comment by nolangrace** [comment]
Was this ever addressed? and if so is it documented anywhere?
